nextroute/schema/input.go

```diff
@@ -31,54 +31,54 @@ type VehicleDefaults struct {
+	UnplannedPenalty        *int       `json:"unplanned_penalty,omitempty" minimum:"0"`
 	Quantity                any        `json:"quantity,omitempty"`
 	StartTimeWindow         any        `json:"start_time_window,omitempty"`
-	MaxWait                 *int       `json:"max_wait,omitempty"`
-	Duration                *int       `json:"duration,omitempty"`
+	MaxWait                 *int       `json:"max_wait,omitempty" minimum:"0"`
+	Duration                *int       `json:"duration,omitempty" minimum:"0"`
 	TargetArrivalTime       *time.Time `json:"target_arrival_time,omitempty"`
-	EarlyArrivalTimePenalty *float64   `json:"early_arrival_time_penalty,omitempty"`
-	LateArrivalTimePenalty  *float64   `json:"late_arrival_time_penalty,omitempty"`
-	CompatibilityAttributes *[]string  `json:"compatibility_attributes,omitempty"`
+	EarlyArrivalTimePenalty *float64   `json:"early_arrival_time_penalty,omitempty" minimum:"0"`
+	LateArrivalTimePenalty  *float64   `json:"late_arrival_time_penalty,omitempty" minimum:"0"`
+	CompatibilityAttributes *[]string  `json:"compatibility_attributes,omitempty" uniqueItems:"true"`
 }
 
 // Vehicle represents a vehicle.
 type Vehicle struct {
 	Capacity                any            `json:"capacity,omitempty"`
 	StartLevel              any            `json:"start_level,omitempty"`
 	CustomData              any            `json:"custom_data,omitempty"`
-	CompatibilityAttributes *[]string      `json:"compatibility_attributes,omitempty"`
-	MaxDistance             *int           `json:"max_distance,omitempty"`
+	CompatibilityAttributes *[]string      `json:"compatibility_attributes,omitempty" uniqueItems:"true"`
+	MaxDistance             *int           `json:"max_distance,omitempty" minimumExclusive:"0"`
 	StopDurationMultiplier  *float64       `json:"stop_duration_multiplier,omitempty"`
 	StartTime               *time.Time     `json:"start_time,omitempty"`
 	EndTime                 *time.Time     `json:"end_time,omitempty"`
 	EndLocation             *Location      `json:"end_location,omitempty"`
-	MinStops                *int           `json:"min_stops,omitempty"`
-	MinStopsPenalty         *float64       `json:"min_stops_penalty,omitempty"`
-	MaxStops                *int           `json:"max_stops,omitempty"`
-	Speed                   *float64       `json:"speed,omitempty"`
-	MaxDuration             *int           `json:"max_duration,omitempty"`
-	MaxWait                 *int           `json:"max_wait,omitempty"`
-	ActivationPenalty       *int           `json:"activation_penalty,omitempty"`
+	MinStops                *int           `json:"min_stops,omitempty" minimum:"0"`
+	MinStopsPenalty         *float64       `json:"min_stops_penalty,omitempty" minimum:"0"`
+	MaxStops                *int           `json:"max_stops,omitempty" minimumExclusive:"0"`
+	Speed                   *float64       `json:"speed,omitempty" minimumExclusive:"0"`
+	MaxDuration             *int           `json:"max_duration,omitempty" minimumExclusive:"0"`
```

Comment:
This can be zero

Reply:
cannot be 0: https://github.com/nextmv-io/platform/blob/2d7a22f5c531adab93c11e146e2d52c3bc3adbf1/core/plugins/nextroute/factory/constraint_max_duration.go#L33

Reply:
We should not throw an error when passing a perfectly feasible limit....